### PR TITLE
fix: implement `CallbackTestRunnerInterface` instead of extending `CallbackTestRunner` class

### PR DESCRIPTION
**Pull request by mrazauskas**
Following up #113

As reported in https://github.com/jest-community/jest-runner-tsd/issues/97 `jest-runner` has to be moved to dependencies instead of peer dependencies. This problem was introduced in #113, because `CallbackTestRunner` class got imported from `'jest-runner'` instead of types only import.

Switching to `CallbackTestRunnerInterface` instead of extending the class should fix the issue.

Closes https://github.com/jest-community/jest-runner-tsd/issues/97

### CODE REVIEW

File: lib/createJestRunner.ts [hunk 1]
```diff
@@ -32,8 +33,14 @@ export default function createRunner<
 >(
   runPath: string,
   { getExtraOptions }: CreateRunnerOptions<ExtraOptions> = {},
-): typeof CallbackTestRunner {
-  return class BaseTestRunner extends CallbackTestRunner {
+) {
+  return class BaseTestRunner implements CallbackTestRunnerInterface {
+    #globalConfig: Config.GlobalConfig;
```
Comment (mrazauskas) [inline lib/createJestRunner.ts:38]: TS is fine with `#`, but does not allow to have `private _globalConfig`: "Property '_globalConfig' of exported class expression may not be private or protected."

Hm.. Should Node 12 support be dropped before merging this change? https://github.com/jest-community/create-jest-runner/blob/445b564d702b085f12a470a4f696745581300df0/.github/workflows/nodejs.yml#L36-L37

Comment (SimenB) [reply lib/createJestRunner.ts:38]: do we need to use `#` now, or is that just a refactoring?

Comment (mrazauskas) [reply lib/createJestRunner.ts:38]: Can be `public` as well. Is that fine?

Comment (SimenB) [reply lib/createJestRunner.ts:38]: Private is fine 🙂 this class isn't used by consumers anyways

Comment (SimenB) [reply lib/createJestRunner.ts:38]: @mrazauskas landed https://github.com/jest-community/create-jest-runner/pull/135 fwiw

**Pull request merged**